narwhal/types/src/failpoints.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use std::env::{self, VarError};
+use tracing::{info, warn};
+
+pub fn initialize_failpoints() {
+    if fail::has_failpoints() {
+        warn!("Failpoints are enabled");
+        let failpoints = match env::var("FAILPOINTS") {
+            Ok(s) => s,
+            Err(VarError::NotPresent) => {
+                // Panic okay here because failpoints feature is set and env var is required to utilize it.
+                panic!("FAILPOINTS environment variable is not set")
+            }
+            Err(e) => {
+                // Panic okay here because failpoints feature is set and valid failpoints are required.
+                panic!("Invalid failpoints: {:?}", e)
+            }
+        };
+        for mut cfg in failpoints.trim().split(';') {
```

Comment:
It seems that this code is mostly copy-paste of FailpointScenario::setup.
Perhaps we should just call that here instead? Something like `mem::forget(FailpointScenario::setup())` to avoid teardown

Reply:
Thanks! This is exactly what I was looking for. I did not want to maintain a handle of the `FailpointScenario` to use `setup()`.